### DISCUSSION TITLE
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

### DISCUSSION THREAD

**Discussion by rudra0713** [PaddlePaddle/PaddleOCR#12959] [Q&A]
Hi,
 I am trying to run the following code:
 ```
python3 table/predict_table.py  --image_dir=/scratch/rrs99/PaddleOCR/ppstructure/page_4.jpg \
                                --det_limit_side_len=736 \
                                --rec_model_dir=/scratch/rrs99/PaddleOCR/ppstructure/inference/en_ppocr_mobile_v2.0_table_rec_infer \
                                --table_model_dir=/scratch/rrs99/PaddleOCR/ppstructure/inference/en_ppocr_mobile_v2.0_table_structure_infer \
                                --det_model_dir=/scratch/rrs99/PaddleOCR/ppstructure/inference/en_ppocr_mobile_v2.0_table_det_infer \
                                --rec_char_dict_path=/scratch/rrs99/PaddleOCR/ppocr/utils/dict/table_dict.txt \
                                --table_char_dict_path=/scratch/rrs99/PaddleOCR/ppocr/utils/dict/table_structure_dict.txt \
                                --det_limit_type=min \
                                --output=/scratch/rrs99/PaddleOCR/ppstructure/output/table
```
However, I am getting the error:
```
Traceback (most recent call last):
  File "/scratch/rrs99/PaddleOCR/ppstructure/table/predict_table.py", line 241, in <module>
    main(args)
  File "/scratch/rrs99/PaddleOCR/ppstructure/table/predict_table.py", line 189, in main
    pred_res, _ = table_sys(img)
  File "/scratch/rrs99/PaddleOCR/ppstructure/table/predict_table.py", line 109, in __call__
    pred_html = self.match(structure_res, dt_boxes, rec_res)
  File "/scratch/rrs99/PaddleOCR/ppstructure/table/matcher.py", line 65, in __call__
    dt_boxes, rec_res = self._filter_ocr_result(pred_bboxes, dt_boxes, rec_res)
  File "/scratch/rrs99/PaddleOCR/ppstructure/table/matcher.py", line 194, in _filter_ocr_result
    y1 = pred_bboxes[:, 1::2].min()
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```
My virtual environment details:
```
paddleocr         2.7.3
paddlepaddle      2.6.1
cuda/12.2 
gcc/12.3 
opencv/4.9.0 
python/3.10
```
Can you please help me, resolve this?

**Comment by Sunting78** [comment]
Could you provide your test images and official links to the various models you used? We would like to replicate the results.

**Comment by rudra0713** [comment]
Hi @Sunting78,
Test image: 
![page_4](https://github.com/PaddlePaddle/PaddleOCR/assets/26213283/98345655-54fd-4bce-9d59-dee8be3a9de0)

And I downloaded the models using the following set of commands
```
wget https://paddleocr.bj.bcebos.com/dygraph_v2.0/table/en_ppocr_mobile_v2.0_table_det_infer.tar && tar xf en_ppocr_mobile_v2.0_table_det_infer.tar

wget https://paddleocr.bj.bcebos.com/dygraph_v2.0/table/en_ppocr_mobile_v2.0_table_rec_infer.tar && tar xf en_ppocr_mobile_v2.0_table_rec_infer.tar

wget https://paddleocr.bj.bcebos.com/dygraph_v2.0/table/en_ppocr_mobile_v2.0_table_structure_infer.tar && tar xf en_ppocr_mobile_v2.0_table_structure_infer.tar
```

**Comment by rudra0713** [comment]
Hi @Sunting78, just checking, in case there is an update on this.

**Comment by adityawardhanm** [comment]
@rudra0713 did you find a solution to this? As I am facing a similar issue. Please tag me if the solution is presented

**Comment by ss756** [comment]
hi was anyone able to debug this issue ?

**Comment by adityawardhanm** [comment]
@ss756 I did encounter this issue and managed to find a workaround. Unfortunately, I don’t have my corrected codebase with me at the moment.

From what I remember, the issue arose when I was using it in a Hindi data stream. To meet my deadline, I ended up removing PaddleOCR’s table detection and used microsoft/table-transformer-detection instead.

If you can share a sample image or more details on your dataset/ usecase, I’d be happy to take a look and help debug the issue further

**Comment by ss756** [comment]
![image](https://github.com/user-attachments/assets/978b6541-da82-403e-9dc8-9f68e56a12a0)
Hi @adityawardhanm  I am running PPStructure v2 and have specified the following table detection model to recognize the table in the image. So instead of using this model you are suggesting to use miscroft's table transformer detection model instead. Am I correct ? Also would it be possible to share a code snippet that showcases this ...

**Reply by adityawardhanm** [reply]
@ss756 I don't have the code on me, as I was working on this code while interning. 

As far as I remember, the reason this happens is that the inbuilt table detection model doesn't function all the time. (i often saw paddleocr paired with other models to extract the most amount of text and to work with tables, lists and other data-structures)

This specific error (according to my hazy memory) is caused by the fact that it doesn't return proper bounding boxes, which cause NaN values to flow in and give the error. If you're working with tables, go with [microsoft/table-transformer-detection](https://huggingface.co/microsoft/table-transformer-detection) (works just fine for english ocr images), this can extract the table structure, and then you can run the simpler ppocr on each cell extracted. Atleast, that's how I did it

**Reply by ss756** [reply]
Thanks @adityawardhanm  bro, definitely will try this !
